packages/components/src/templates/next/components/internal/Siderail/SiderailLabel.tsx

```diff
@@ -0,0 +1,57 @@
+import { BiRightArrowAlt } from "react-icons/bi"
+import { tv } from "tailwind-variants"
+
+import type { Item } from "./types"
+import { useSiderailContext } from "./SiderailContext"
+
+const createSiderailLabelStyles = tv({
+  slots: {
+    container:
+      "group flex gap-1.5 px-3 py-4 transition hover:text-brand-interaction",
+    icon: "h-6 w-6 opacity-0 transition group-hover:translate-x-1 group-hover:opacity-100",
+  },
+  variants: {
+    isCurrent: {
+      true: {
+        container: "text-brand-interaction",
+      },
+    },
+  },
+})
+
+const compoundStyles = createSiderailLabelStyles()
+
+interface SiderailLabelProps extends Omit<Item, "childPages"> {
+  className?: string
+  showIconOnHover?: boolean
+}
+
+export const SiderailLabel = ({
+  isCurrent,
+  title,
+  url,
+  className,
+  showIconOnHover,
+}: SiderailLabelProps) => {
+  const { LinkComponent } = useSiderailContext()
+  if (isCurrent) {
+    return (
+      <p className={compoundStyles.container({ className, isCurrent })}>
+        {title}
+      </p>
+    )
+  }
+
+  return (
+    <LinkComponent
+      href={url}
+      className={compoundStyles.container({ className })}
+    >
+      {title}
+
+      {showIconOnHover && (
+        <BiRightArrowAlt aria-hidden className={compoundStyles.icon()} />
+      )}
```

Comment:
issue: the arrow's positioning on the user's browser doesn't seem to be consistent. One is beside the text and the other is at the end of the siderail. I think we should just put the arrow right at the end?

https://github.com/user-attachments/assets/03516088-68a4-4d98-8a51-e97ab80167d4